### PR TITLE
test(add-missing-tests ): add the missing tests

### LINKED ISSUE #13: Improve test case coverage for Java core lib
Improve test case coverage for java core lib by adding/improving test cases.

### PR DESCRIPTION
**Pull request by hamzashoukat94**
Add the missing tests to increase the code coverage of the core library.

closes #13

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -66,6 +66,11 @@
 			<artifactId>jaxb-api</artifactId>
 			<version>2.4.0-b180830.0359</version>
 		</dependency>
+		<dependency>
+			<groupId>org.glassfish.jaxb</groupId>
+			<artifactId>jaxb-runtime</artifactId>
```
Comment (sufyankhanrao) [inline pom.xml:71]: Question: Why is this dependency needed?

File: src/main/java/io/apimatic/core/types/http/response/Dynamic.java [hunk 2]
```diff
@@ -138,7 +135,6 @@ public double parseAsDouble() throws ParseException {
      * Parse response as string.
      * 
      * @return Parsed value
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/core/types/http/response/Dynamic.java:137]: ParseException description should also be reverted in the method summary.

File: src/main/java/io/apimatic/core/types/BaseModel.java [hunk 3]
```diff
@@ -148,9 +148,12 @@ private Method getMethodMatchingKey(String key, boolean getter) {
                 annotation = method.getAnnotation(JsonSetter.class);
             }
             if (annotation != null) {
-                if ((getter) && ((JsonGetter) annotation).value().equalsIgnoreCase(key)) {
+                String methodName = key;
+                if ((getter) && ((JsonGetter) annotation).value()
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/core/types/BaseModel.java:152]: Nits, Non-blocker: Extra round brackets can be removed on the `getter` conditional arg.

File: src/main/java/io/apimatic/core/utilities/CoreHelper.java [hunk 4]
```diff
@@ -1187,7 +1187,6 @@ private static String getAccessorStringFormat(
                 return "%7C";
             case TSV:
                 return "%09";
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/core/utilities/CoreHelper.java:1189]: Nits, Non-blocker: Use proper constant variable name for the hardcoded strings for CSV, PSV and TSV to increase the readability.

File: pom.xml (hunk 1 above)
Comment (hamzashoukat94) [reply pom.xml:71]: It is required to resolve the JAXB dependency. It was facing this [issue](https://stackoverflow.com/questions/51916221/javax-xml-bind-jaxbexception-implementation-of-jaxb-api-has-not-been-found-on-mo).

File: src/main/java/io/apimatic/core/types/CoreApiException.java [hunk 5]
```diff
@@ -60,7 +60,10 @@ public CoreApiException(String reason, Context context) {
      * @return Returns the response code for ApiException
      */
     public int getResponseCode() {
-        return (httpContext != null) ? httpContext.getResponse().getStatusCode() : -1;
+        return (httpContext != null)
+                ? (httpContext.getResponse() != null ? httpContext.getResponse().getStatusCode()
```
[the remark this answers is not in the document]
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/core/types/CoreApiException.java]: Used early return

File: src/main/java/io/apimatic/core/utilities/CoreHelper.java (hunk 4 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/core/utilities/CoreHelper.java]: Added the proper constants for hardcoded strings

File: src/main/java/io/apimatic/core/types/BaseModel.java (hunk 3 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/core/types/BaseModel.java]: Removed extra brackets

File: src/main/java/io/apimatic/core/types/http/response/Dynamic.java (hunk 2 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/core/types/http/response/Dynamic.java:137]: Reverted

File: src/main/java/io/apimatic/core/types/http/response/Dynamic.java [hunk 6]
```diff
@@ -136,9 +133,8 @@ public double parseAsDouble() throws ParseException {
 
     /**
      * Parse response as string.
-     * 
-     * @return Parsed value
      * @throws ParseException Signals if a parse exception occured
+     * @return Parsed value
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/core/types/http/response/Dynamic.java:137]: Nits, Non-blocker: `@throws` should be the last line of the method summary.

Comment (hamzashoukat94) [reply src/main/java/io/apimatic/core/types/http/response/Dynamic.java:137]: `@throws` Moved at the end of the method summary

**Review by sufyankhanrao — APPROVED** [review]
Changes look good to me.

**Pull request merged**